src/dataregistry/registrar_util.py

```diff
@@ -1,30 +1,44 @@
 import os
+import re
+from sqlalchemy import MetaData, Table, Column, text, select
+
 from dataregistry.db_basic import ownertypeenum
 
-__all__ = ['make_version_string', 'parse_version_string']
+__all__ = ['make_version_string', 'parse_version_string', 'calculate_special',
+           'form_dataset_path', 'get_directory_info']
 VERSION_SEPARATOR = '.'
+_nonneg_int_re = "0|[1-9][0-9]*"
+
 def make_version_string(major, minor=0, patch=0, suffix=None):
```

Comment:
I don't see this function being used anywhere.

Also it would need to be `version = VERSION_SEPARATOR.join([str(major), str(minor), str(patch)])`

Reply:
Likely  not.  It was used with the old interface to `register_dataset`, when `major`, `minor`, `patch` were separate arguments.  Good catch.